### PR TITLE
GMC: Add GPU support for GMC.

### PR DESCRIPTION
**Pull request by PeterYang12**
## Description

Enable NVIDIA GPU support for GMC, including sequence and switch modes. Note that switch mode may fail due to not enough GPU memory.

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

File: microservices-connector/config/samples/chatQnA_switch_nv.yaml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            config:
+              endpoint: /generate
+              MODEL_ID: Intel/neural-chat-7b-v3-3
+            isDownstreamService: true
+        - name: TgiNvidia
+          internalService:
+            serviceName: tgi-service-llama
+            config:
+              endpoint: /generate
+              MODEL_ID: Intel/neural-chat-7b-v3-3
```
Comment (irisdingbj) [inline microservices-connector/config/samples/chatQnA_switch_nv.yaml:123]: this is supposed to be the llama model

File: microservices-connector/usage_guide.md [hunk 2]
```diff
@@ -39,7 +43,7 @@ kubectl create deployment client-test -n chatqa --image=python:3.8.13 -- sleep i
 **Access the pipeline using the above URL from the client pod**
 
 ```bash
-export CLIENT_POD=$(kubectl get pod  -l app=client-test -o jsonpath={.items..metadata.name})
+export CLIENT_POD=$(kubectl get pod -n chatqa  -l app=client-test -o jsonpath={.items..metadata.name})
```
Comment (irisdingbj) [inline microservices-connector/usage_guide.md:46]: good catch!

File: microservices-connector/config/samples/chatQnA_switch_nv.yaml (hunk 1 above)
Comment (PeterYang12) [reply microservices-connector/config/samples/chatQnA_switch_nv.yaml:123]: Thank you for your review. Here is a workaround. If using the llama model, one GPU card may fail to launch two instance due to few GPU memory. I am ok to keep the llama model here, what do you think?

Comment (irisdingbj) [reply microservices-connector/config/samples/chatQnA_switch_nv.yaml:123]: this is just to work around for current NV machine, for the example we want to provide to end user, it is better we use a meaningful example, like keep the llama model.

Comment (PeterYang12) [reply microservices-connector/config/samples/chatQnA_switch_nv.yaml]: > this is just to work around for current NV machine, for the example we want to provide to end user, it is better we use a meaningful example, like keep the llama model.

Fixed

**Review by irisdingbj — APPROVED** [review]

**Review by daisy-ycguo — APPROVED** [review]
lgtm

**Comment by PeterYang12** [comment]
Thank you, Iris and Daisy. :)

**Pull request merged**